caas/kubernetes/provider/bootstrap.go

```diff
@@ -97,6 +123,11 @@ func newcontrollerStack(stackName string, storageClass string, broker *kubernete
 	agentConfig.SetStateServingInfo(si)
 	pcfg.Bootstrap.StateServingInfo = si
 
+	// we use controller name to name controller namespace in bootstrap time.
+	if err = setControllerNamespace(pcfg.ControllerName, broker); err != nil {
```

Comment:
We know what the namespace will be very early on - the broker should be constructed with the correct namespace initially. I don't think we should have this method.

Reply:
for a new/non-existing controller, here is the first time to know the controllerName.